### PR TITLE
CID-1136---add-new-id-logic

### PR DESCRIPTION
**Pull request by vg-leanix**
this PR essentially feeds the GitHub repo id as id for the vsm-collector api. This allows better handling in VSM. For example if services from GitHub and GitHub action need to be mapped automatically, the repository id can be used as such

### CODE REVIEW

File: src/mtm.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ async function authenticate(host, token) {
     console.info(`Successfully generated JWT token.`);
 
     return axios.create({
-      baseURL: `https://eu-vsm.leanix.net/services/vsm/discovery/v1`,
```
Comment (vg-leanix) [inline src/mtm.js:23]: @aravindmetku as hinted in the comment. I believe this should be parametrized depending on the host. Personally, I'd even expect the following:
1. call MTM with API token 
2. get bearer token
3. decode bearer token and parse `iss` attribute to capture the instance. e.g. eu/us...
4. us region from 3 as input to call vsm-collector

this makes it as user friendly as possible and doing the heavy lifting on the backend.

Comment (vg-leanix) [reply src/mtm.js:23]: image

Comment (aravindmetku) [reply src/mtm.js]: thank you Vince for the insight. The suggestions are incorporated!

File: src/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const fs = require("fs");
 
 const { authenticate } = require("./mtm");
 const { registerService } = require("./discovery-api-client");
+const { validateInputs } = require("./validations");
```
Comment (alfredo-mfaria) [inline src/index.js:6]: I can't write this comment on the correct line, as this is not part of this PR, but why do we do this:
let dryRun = core.getInput("dry-run");
dryRun = !(dryRun === "false");

**Review by alfredo-mfaria — COMMENTED** [review]
Shall we add sanitisation on the "host" parameter?
we expect the user to copy/paste the url, I would expect some to include "https://",  or something like "https://leanix.leanix.net/lxvsm" or even blank spaces

File: src/index.js (hunk 2 above)
Comment (aravindmetku) [reply src/index.js:6]: it's funny, the input from yaml is not parsed correctly by github core package. it comes as string. any non-empty string is a truly value in JS/TS so we added this logic

**Comment by aravindmetku** [comment]
@alfredo-mfaria yes, good point, I have added sanitise logic for host 
sanistise is only removing blank spaces

**Review by alfredo-mfaria — APPROVED** [review]

**Pull request merged**